### DISCUSSION TITLE
Project build is failing.

### DISCUSSION THREAD

**Discussion by kycasdzxc** [yayakm/RegexStringGenerator#21] [Q&A] [answered]
I downloaded the source code locally for project analysis, but I'm encountering the following error and cannot build the project. Isn't this project a Java & Maven project? Are there any additional configurations I need to consider?

```
Project build error: Non-resolvable parent POM: Failure to transfer org.springframework.boot:spring-boot-starter-parent:pom:2.7.12 from https://repo.maven.apache.org/maven2 
 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced. Original error: Could not transfer artifact 
 org.springframework.boot:spring-boot-starter-parent:pom:2.7.12 from/to central (https://repo.maven.apache.org/maven2): Network is unreachable: connect and 'parent.relativePath' 
 points at no local POM
```
![image](https://github.com/user-attachments/assets/942348a4-861e-4b87-9492-f174ab6c0528)
![image](https://github.com/user-attachments/assets/f60c5417-8e9d-4936-9b0f-3cb52e6a3d99)

**Comment by bak840** [comment, marked as answer]
hi, managed to run the project on IntelliJ IDEA.The IDE managed to fetch the dependencies without any issues. Are you using a VPN, a proxy or any specific network configuration?

**Reply by yayakm** [reply]
@kycasdzxc 

> . Are you using a VPN, a proxy or any specific network configuration?

It might be a problem with the internet connection.  Check whether you need to configure a proxy server.
Here's the documentation for configuring a maven proxy server: https://maven.apache.org/guides/mini/guide-proxies.html

If the problem persists, try to update maven dependencies by hands `mvn dependency:purge-local-repository` or by deleting the folder XXX\.m2\repository\org\springframework\boot\spring-boot-parent and then re-importing the project.

**Reply by kycasdzxc** [reply]
> hi, managed to run the project on IntelliJ IDEA.The IDE managed to fetch the dependencies without any issues. Are you using a VPN, a proxy or any specific network configuration?

Solved. It seems that the company has a proxy. It works well because I do it at home. thank you.

![image](https://github.com/user-attachments/assets/e359ffb4-3b8f-4da8-8b0c-f9a2978eb45a)
